libsignal-service/src/sender.rs

```diff
@@ -645,23 +646,20 @@ where
                 )
                 .await?,
             );
+        } else {
+            devices.push(DEFAULT_DEVICE_ID);
         }
 
-        let mut devices = vec![DEFAULT_DEVICE_ID];
         devices
             .extend(recipient.sub_device_sessions(&self.session_store).await?);
+        devices.dedup();
```

Comment:
dedup only works for consecutive duplicates, and should therefor come after the `sort_unstable`. Is there a reason you altered that?

Reply:
I wasn't aware that `dedup()` works that way; I'll change it back the way it was.